tamr_sdk/utils/handler.py

```diff
@@ -0,0 +1,17 @@
+import grpc
+
+def exception_handler(func):
+    def handle_grpc_status_codes(*args, **kwargs):
+        try:
+            func(*args, **kwargs)
+        except grpc._channel._InactiveRpcError as e:
+            if e.code() == grpc.StatusCode.NOT_FOUND:
+                raise ValueError(e.details())
```

Comment:
Any thoughts on making all these `raise <Error> from None` ? 
That makes the output when an error is caught cleaner (but maybe less informative) -- e.g. without the `from None`, the error looks like 
```
Traceback (most recent call last):
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/tamr_sdk/utils/handler.py", line 17, in handle_grpc_status_codes
    func(_self, *args, **kwargs)
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/tamr_sdk/jobs/jobs_client.py", line 17, in get_job
    return self.stub.GetJob(request, metadata=self.metadata)
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/.venv/lib/python3.9/site-packages/grpc/_channel.py", line 1176, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/.venv/lib/python3.9/site-packages/grpc/_channel.py", line 1005, in _end_unary_response_blocking
    raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.NOT_FOUND
        details = "Job not found: <job-id>"
        debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2024-05-07T08:49:12.737176-04:00", grpc_status:5, grpc_message:"Job not found: <job-id>"}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/examples/list_jobs.py", line 15, in <module>
    response = jobs.get_job(job_id="<job-id>")
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/tamr_sdk/utils/handler.py", line 20, in handle_grpc_status_codes
    raise ValueError(e.details())
ValueError: Job not found: <job-id>
```
whereas with the `from None`, it looks like 
```
Traceback (most recent call last):
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/examples/list_jobs.py", line 15, in <module>
    response = jobs.get_job(job_id="<job-id>")
  File "/Users/erin.compaan/Repos/tamr-cloud-sdk-python/tamr_sdk/utils/handler.py", line 20, in handle_grpc_status_codes
    raise ValueError(e.details()) from None
ValueError: Job not found: <job-id>
```

Reply:
Hmmm interesting. There is definitely a trade off here. I think I slightly lean towards being more informative and not excluding information, but I do see that the second is more readable and in most cases the grpc error is not going to be informative. What do you think?

Reply:
I'm not sure what the wrapper is accomplishing if it's just adding another error to the stack of errors thrown by gRPC (other than the potential to add different handling later). But I agree it would potentially be aggravating to hit an error and be unable to access the raw information about what went wrong. I would like to make it configureable, but not sure how to do that in the context of a wrapper. Perhaps using an environment variable?

Reply:
The idea was that it is adding a more pythonic error code. I realize in these simple cases it is not very helpful, but the hope would be as we add more complexity having the pythonic error codes may be helpful. I will give making it configurable a shot

Reply:
Okay, I think the most recent push makes it configurable